Add support for DescendantProviderInterface for pages

Prokyonn opened the pull request:
| Q | A
  | --- | ---
  | Bug fix? | no
  | New feature? | yes
  | BC breaks? | no
  | Deprecations? | no
  | Related issues/PRs | #7672 
  | License | MIT

  #### What's in this PR?

  This PR implements `DescendantProviderInterface` support for pages, adding:
  - Permission checks for all descendant pages when deleting a parent page
  - Warning dialog when attempting to delete pages with children (requires `force=true` to proceed)
  - `findDescendantIdsById()` method in PageRepository for querying page hierarchies

  #### Why?

Missing feature from Sulu 3.0

Prokyonn committed:
Add DescendantProviderInterface support for pages

alexander-schranz commented on packages/page/src/Application/MessageHandler/RemovePageMessageHandler.php:
```diff
@@ -28,9 +37,16 @@
+        private AccessControlRepositoryInterface $accessControlRepository,
+        private SystemStoreInterface $systemStore,
+        private ?Security $security,
+        private array $permissions,
```
Unsure handle this in the application layer as its unrelated to page domain, as security is more related to UserInterface or Infrastracture Layer 🤔 .

Prokyonn replied:
> Unsure handle this in the application layer as its unrelated to page domain, as security is more related to UserInterface or Infrastracture Layer 🤔 .

Yeah, I wasn’t completely sure about it either. It also feels a bit odd in the controller.

alexander-schranz replied:
as discussed we move to request listener

Prokyonn committed:
Add Review suggestions

# Conflicts:
#	packages/page/src/Infrastructure/Symfony/HttpKernel/SuluPageBundle.php
